package/src/memmon.cpp

```diff
@@ -75,3 +76,40 @@ std::map<std::string, double> const memmon::get_json_average_stats(
     unsigned long long elapsed_clock_ticks) {
   return mem_average_stats;
 }
+
+// Collect related hardware information
+std::map<std::string, std::map<std::string, std::string>> const memmon::get_hardware_info() {
+  std::map<std::string, std::map<std::string, std::string>> result{};
+
+  // Read some information from /proc/meminfo
+  std::ifstream memInfoFile{"/proc/meminfo"};
+  if(!memInfoFile.is_open()) {
+    std::cerr << "Failed to open /proc/meminfo" << std::endl;
+    return result;
+  }
+
+  // Metrics to read from the input
+  std::vector<std::string> metrics{"MemTotal"};
+
+  // Loop over the file
+  std::string line;
+  while (std::getline(memInfoFile,line)) {
+    if (line.empty()) continue;
+    size_t splitIdx = line.find(":");
+    std::string val;
+    if (splitIdx != std::string::npos) {
+      val = line.substr(splitIdx + 1);
+      if (val.empty()) continue;
+      for (const auto& metric : metrics) {
+        if (line.size() >= metric.size() && line.compare(0, metric.size(), metric) == 0) {
+          result["mem"][metric] = std::regex_replace(val, std::regex("^\\s+|\\s+$"), "");
```

Comment:
I would report this without the `kB` units in it (consistency with the way the rest of prmon works, Taking into account the above comment, it should be a JSON number, not a string.

Reply:
Done.